fastlane/lanes/build.rb

```diff
@@ -134,6 +134,27 @@
     end
   end
 
+  #####################################################################################
+  # download_signed_apk_from_google_play
+  # -----------------------------------------------------------------------------------
+  # This lane downloads the signed apk from Play Store for the given version and app
+  # -----------------------------------------------------------------------------------
+  # Usage:
+  # bundle exec fastlane download_signed_apk_from_google_play app:<wordpress|jetpack> version:<versionName,versionCode>
```

Comment:
I don't see from the code how passing `version:12.3,789` would allow you to provide the version name and code in one comma-separated parameter like how it's suggested in this doc?

Reply:
I actually tried to figure out how we were supposed to pass a Hash from command line and couldn't do it. So, I copied this documentation format from `build_bundle` lane. I figured, maybe this is how we are documenting it for when the value is passed from another lane.

I'd appreciate it if you can share the appropriate way to handle this. Thanks!

Reply:
Yeah unfortunately, to my knowledge at least, there's no easy way to provide a Hash value via command line.

From what I see, [the way this has been handled in other lanes is by explicitly testing if `version` is a `String` after pulling its value from `options[:version]`, and if so then split the string with `,` and reconstruct the `Hash`](https://github.com/wordpress-mobile/WordPress-Android/blob/bb4eb2d39f6ff2bf95f53c337bf2bb12a03f9f72/fastlane/lanes/build.rb#L203-L206).
```ruby
    if version.is_a?(String) # for when calling from command line
      (version_name, version_code) = version.split(',')
      version = { 'name' => version_name, 'code' => version_code || '1' }
    end
```

---

The alternative solution would be the other way around, i.e. have the lane take the options separately (`options[:versionName]` and `options[:versionCode]`), so you can provide separate String values from the command line; and then, for when you call the lane via another lane which has the `version` in a variable as a `Hash` (from calling `version = android_get_release_version()`), we'd have to explicitly pass `versionName: version['name'], versionCode: version['code']` as separate parameters…

Up to you. But given other lanes are using the `if version.is_a?(String) … ` trick so far, I guess you it would make sense to continue that way for your new lane here?

Reply:
Thanks a lot for all this explanation @AliSoftware! I should have checked if the `version` was handled in a different way for these lanes, but I am really not used to this kind of pattern - as they are impossible in typed languages - so I didn't expect it at all.

I've made the changes in dbe95dfb2305334fa4aed88034554ea8b23d5223 and ran some random tests that seems to work as expected:

```
bundle exec fastlane download_signed_apk_from_google_play app:wordpress version:22.9-rc-1,1359
bundle exec fastlane download_signed_apk_from_google_play app:wordpress version:22-rc-1,1359
bundle exec fastlane download_signed_apk_from_google_play app:jetpack version:arst,1300
```

Reply:
> but I am really not used to this kind of pattern - as they are impossible in typed languages

Oh yeah I'm with you on that, this is not super pretty. Which is why I also mentioned the second option / alternative solution to pass those as separate options to the lane — so command line calls can pass them separately, while parent lanes calling that lane can pass each entry of the Hash separately — which I feel would be much cleaner to understand and would avoid us to rely on having to test the type of the variable with `is_a?(String)`!

So feel free to adopt this alternative solution at some point in a subsequent PR, to clean up the logic and implementation if you want to make this less confusing / convoluted / less relying on Ruby duck-typing.

Reply:
TIL: after randomly browsing the source code for fastlane's `ConfigItem` class, I just discovered [that fastlane _does_ support passing a Hash from the command line… as long as you use valid JSON syntax](https://github.com/fastlane/fastlane/blob/master/fastlane_core/lib/fastlane_core/configuration/config_item.rb#L283-L288) 🤯   Didn't know that.

That being said, this would make the command line call quite ugly to write for that case… so it's probably not the direction we want to go to allow this lane to be called from CLI 🙃

Reply:
I could swear that I've seen it before _somewhere_, but it might still be a delusion. Anyhow, thank you for finding this!